pkg/user/user_integration_test.go

```diff
@@ -303,6 +591,30 @@ func TestUserHandler(t *testing.T) {
 	})
 }
 
+type userService interface {
+	FindById(context context.Context, id uint) (*model.User, error)
+	ValidateEmail(emailToken uuid.UUID) error
+}
+
+func createUser(t *testing.T, client *inttest.HTTPClient, userService userService, email string, password string) *model.User {
```

Comment:
You are returning a user but not using it. You could generate the `email` and 
`password` in here, save 2 params and return and use the user to create the token in the requests.

Maybe use a uuid as in `uuid@dhis2.org` for email and password.

It might even be nice to create a struct which has an https://pkg.go.dev/sync/atomic#Int32 field. Then this method increments it on every call. This would be nice as I see we have one test that asserts how many users are returned which is a magic number.

Reply:
Please see the updated version